databricks/koalas/series.py

```diff
@@ -3223,6 +3223,65 @@ def copy(self) -> 'Series':
         """
         return _col(DataFrame(self._internal.copy()))
 
+    def first_valid_index(self):
+        """
+        Retrieves the index of the first valid value.
+
+        Returns
+        -------
+        idx_first_valid : type of index
+
+        Examples
+        --------
+        >>> s = ks.Series([None, None, 3, 4, 5], index=[100, 200, 300, 400, 500])
+        >>> s
+        100    NaN
+        200    NaN
+        300    3.0
+        400    4.0
+        500    5.0
+        Name: 0, dtype: float64
+
+        >>> s.first_valid_index()
+        300
+
+        Support for MultiIndex
+
+        >>> midx = pd.MultiIndex([['lama', 'cow', 'falcon'],
+        ...                       ['speed', 'weight', 'length']],
+        ...                      [[0, 0, 0, 1, 1, 1, 2, 2, 2],
+        ...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
+        >>> s = ks.Series([None, None, None, None, 250, 1.5, 320, 1, 0.3], index=midx)
+        >>> s
+        lama    speed       NaN
+                weight      NaN
+                length      NaN
+        cow     speed       NaN
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.first_valid_index()
+        ('cow', 'weight')
+        """
+        sdf = self._internal.sdf
+        data_scol = self._internal.data_scols[0]
+
+        first_valid_value = sdf.select(
+            F.first(data_scol, True).alias(str(self.name))).first()[str(self.name)]
```

Comment:
This can't guarantee the first value since `F.first()` is non-deterministic.

Reply:
@ueshin Thanks for reviewing, but could you explain a little bit more detail?

below is the point why i hard to understand `F.first()` is non-deterministic

if `koalas.Series` looks like below.

```python
>>> s = ks.Series([None, 1, 2, 3, 4], index=[4, 5, 2, 1, 8])
>>> s
4    NaN
5    1.0
2    2.0
1    3.0
8    4.0
Name: 0, dtype: float64
```

then, the internal spark dataframe looks like below.

```
>>> s._internal.sdf.show()
+-----------------+----+
|__index_level_0__|   0|
+-----------------+----+
|                4|null|
|                5| 1.0|
|                2| 2.0|
|                1| 3.0|
|                8| 4.0|
+-----------------+----+
```

for this `sdf`, isn't the `F.first()` returns always same result? (in other words, isn't the `F.first()` is deterministic for same `sdf`?)

```
>>> sdf.select(F.first('0', True)).show()
+--------------+
|first(0, true)|
+--------------+
|           1.0|
+--------------+
```

Reply:
I think he meant the dataframe order itself is not guaranteed and therefore the first value can be non-deterministic. If this was correct, not sure what's the right solution for this case. Indexes can have its own natural order and it's lost.

Maybe we should explicitly note that Koalas DataFrame currently does not keep its natural order and it might cause unexpected results (and maybe ti should explicitly sort).

Reply:
The `first` function doc says:

```
    .. note:: The function is non-deterministic because its results depends on order of rows which
        may be non-deterministic after a shuffle.
```

Reply:
oh i got it. i just fixed to not use `F.first()`